src/modules/nft/index.js

```diff
@@ -0,0 +1,132 @@
+import Nft from './src';
+export default class NFT {
+  constructor({ network, address, web3, errorHandler }) {
+    this.network = network;
+    this.address = address;
+    this.web3 = web3;
+    this.errorHandler = errorHandler || console.error;
+    this.nft = new Nft({
+      network: this.network,
+      address: this.address,
+      tokenSetUpdateHook: this.updateActiveTokenSet.bind(this),
+      setAvailableContracts: this.setAvailableContracts.bind(this),
+      web3: this.web3,
+      errorHandler: this.errorHandler
+    });
+    this.availableContracts = [];
+    this.selectedContract = {};
+    this.activeTokenSet = [];
+    this.currentActive = '';
+    this.currentPageState = '';
+    this.activateResolver = '';
+    this.ready = false;
+  }
+
+  async init(selectedContractOverride) {
+    try {
+      let selectedContract = await this.nft.setup();
+      if (selectedContractOverride) {
+        selectedContract = selectedContractOverride;
+      }
+      this.selectedContract = selectedContract;
+      this.currentActive = await this.nft.getFirstTokenSet(selectedContract);
+      this.currentPageState = this.currentActive.getPageState();
+      this.ready = true;
+    } catch (e) {
+      this.errorHandler(e);
+    }
+  }
+
+  get currentActiveContract() {
+    return this.currentActive.contract;
+  }
+
+  send(to, tokenId) {
+    return this.currentActive.send(to, tokenId);
+  }
+
+  removeSentNft(tokenId) {
+    this.currentActive.removeSentNft(tokenId);
+  }
+
+  getAvailableContracts() {
+    return this.nft.getOwnedTokenBasicDetails();
+  }
+
+  setActiveContract(contractAddress) {
+    return new Promise(resolve => {
+      // if (!contractAddress) return this.selectedContract;
+      if (!this.nft.nftConfig[contractAddress]) {
+        this.errorHandler('No NFTs found for contract address');
+        resolve();
+      }
+      this.activateResolver = resolve;
+      this.currentActive = this.nft.nftConfig[contractAddress];
+      this.nft.nftConfig[contractAddress].activate();
```

Comment:
lets return this promise and not set it on this.activateResolver 
init() can return a promise as well